app/src/androidTest/java/com/example/pantrypal/LoginTest.java

```diff
@@ -0,0 +1,102 @@
+package com.example.pantrypal;
+
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.espresso.Espresso;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.assertion.ViewAssertions;
+import androidx.test.espresso.matcher.ViewMatchers;
+
+import android.text.InputType;
+
+import static androidx.test.espresso.Espresso.*;
+import static androidx.test.espresso.action.ViewActions.*;
+import static androidx.test.espresso.assertion.ViewAssertions.*;
+import static androidx.test.espresso.matcher.ViewMatchers.*;
+
+import static java.util.EnumSet.allOf;
+import static org.hamcrest.Matchers.*;
+
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.filters.LargeTest;
+import androidx.test.runner.AndroidJUnit4;
+
+import com.example.pantrypal.Login;
+import com.example.pantrypal.MainActivity;
+import com.example.pantrypal.R;
+
+import org.junit.*;
+
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
```

Comment:
The `AndroidJUnit4` used here is deprecated. Probably shouldn't use it then. Fortunately there seems to be a drop-in replacement for it, so it should only require changing an import statement (maybe some change to the Gradle file?). Take a look here: https://developer.android.com/reference/androidx/test/runner/AndroidJUnit4